solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/batch/SolrSearchSink.java

```diff
@@ -0,0 +1,151 @@
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.output.NullOutputFormat;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Batch SolrSearch Sink Plugin - Writes data to a Sinlge Node Solr or to SolrCloud.
+ */
+@Plugin(type = BatchSink.PLUGIN_TYPE)
+@Name("SolrSearch")
+@Description("This plugin allows users to build the pipelines to write data to Solr. " +
+  "The input fields coming from the previous stage of the pipeline are mapped to Solr fields. User can also specify " +
+  "the mode of the Solr to connect to. For example, Single Node Solr or SolrCloud.")
+public class SolrSearchSink extends BatchSink<StructuredRecord, Text, Text> {
+  private static final Gson GSON = new Gson();
+  private final BatchSolrSearchConfig batchConfig;
+  private Boolean flag = true;
+
+  public SolrSearchSink(BatchSolrSearchConfig batchConfig) {
+    this.batchConfig = batchConfig;
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext context) throws Exception {
+    Job job = context.getHadoopJob();
+    job.setOutputFormatClass(NullOutputFormat.class);
+    context.addOutput(Output.of(batchConfig.referenceName, new SolrSearchSink.SolrOutputFormatProvider(batchConfig)));
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    batchConfig.validateSolrConnectionString();
+    if (inputSchema != null) {
+      batchConfig.validateKeyField(inputSchema);
+      batchConfig.validateInputFieldsDataType(inputSchema);
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord structuredRecord, Emitter<KeyValue<Text, Text>> emitter) throws Exception {
+    Type schemaType = new TypeToken<Schema>() { }.getType();
+    if (flag) {
+      batchConfig.verifySolrConfiguration();
+      flag = false;
+    }
+    batchConfig.validateKeyField(structuredRecord.getSchema());
+    batchConfig.validateInputFieldsDataType(structuredRecord.getSchema());
+
+    if (structuredRecord.get(batchConfig.getKeyField()) == null) {
+      return;
+    }
+    emitter.emit(new KeyValue<Text, Text>(new Text(GSON.toJson(structuredRecord.getSchema(), schemaType)),
+                                          new Text(StructuredRecordStringConverter.toJsonString(structuredRecord))));
+  }
+
+  /**
+   * Output format provider for Batch SolrSearch Sink.
+   */
+  private static class SolrOutputFormatProvider implements OutputFormatProvider {
+    private Map<String, String> conf;
+
+    SolrOutputFormatProvider(BatchSolrSearchConfig batchConfig) {
+      this.conf = new HashMap<>();
+      conf.put("solr.server.url", batchConfig.getSolrHost());
+      conf.put("solr.server.mode", batchConfig.getSolrMode());
+      conf.put("solr.server.collection", batchConfig.getCollectionName());
```

Comment:
you've defined constants for these keys in another class.
Why not make them public and use those constants here, instead of using the string?